packages/dom/src/dom/document-has-text-selection.js

```diff
@@ -10,7 +15,9 @@
  * @return {boolean} True if there is selection, false if not.
  */
 export default function documentHasTextSelection( doc ) {
+	assertIsDefined( doc.defaultView, 'doc.defaultView' );
 	const selection = doc.defaultView.getSelection();
+	assertIsDefined( selection, 'selection' );
 	const range = selection.rangeCount ? selection.getRangeAt( 0 ) : null;
-	return range && ! range.collapsed;
+	return !! range && ! range.collapsed;
```

Comment:
`range` is not a boolean so we cast it to a boolean to appease the return value.